radixdlt/src/test/java/com/radixdlt/sanitytestsuite/scenario/jsonserialization/ArgumentsExtractor.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.sanitytestsuite.scenario.jsonserialization;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import com.radixdlt.atommodel.tokens.MutableSupplyTokenDefinitionParticle;
+import com.radixdlt.atommodel.tokens.TokenPermission;
+import com.radixdlt.identifiers.RRI;
+import com.radixdlt.identifiers.RadixAddress;
+import com.radixdlt.utils.Pair;
+import com.radixdlt.utils.UInt256;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+public final class ArgumentsExtractor {
+
+	private final Set<String> fieldsExtracted;
+	private final JsonNode argumentsObject;
+
+	ArgumentsExtractor(JsonNode argumentsObject) {
+		this.fieldsExtracted = new HashSet<>();
+		this.argumentsObject = argumentsObject;
+	}
+
+	<T> T extractArgument(String fieldNameInJsonObject, Function<JsonNode, T> mapper) {
+		if (fieldsExtracted.contains(fieldNameInJsonObject)) {
+			throw new RuntimeException("Field with name " + fieldNameInJsonObject + " already extracted.");
```

Comment:
Should probably not be a runtime exception but something else and this method should probably cast..